lib/matplotlib/tests/test_datetime.py

```diff
@@ -446,11 +446,20 @@ def test_plot_date(self):
         ax2.plot_date(x_dates, y_ranges)
         ax3.plot_date(x_ranges, y_dates)
 
-    @pytest.mark.xfail(reason="Test for quiver not written yet")
     @mpl.style.context("default")
     def test_quiver(self):
-        fig, ax = plt.subplots()
-        ax.quiver(...)
+        dates = [datetime.datetime(2023, 1, 1) +
+                    datetime.timedelta(days=i) for i in range(5)]
+        x = dates
+        y = np.zeros_like(x, dtype=float)
+        u = np.sin(np.arange(len(x)))
+        v = np.cos(np.arange(len(x)))
+        fig, ax = plt.subplots(figsize=(10, 6))
+        ax.quiver(x, y, u, v, scale=20)
```

Comment:
It seems like `y` could also be tested? I'm not sure if it also makes sense to try `u` and `v` as well.

Reply:
Quiver plots typically expect numerical values for u and v (representing vector components), and y is usually a numerical position value. Using datetime objects for these variables would be unconventional and might not work directly with the quiver plot functionality. I think it might be better to maintain the test as is.

Reply:
I don't really follow the assertion that `y` is usually a numerical position value. But in any case, these tests are to check whether the code works, not what the "usual" type is. We absolutely want to test things that are unusual.

Reply:
I agree that y is not actually different from x here and should be tested, additionally I _do_ think u/v should be tested with `angles="xy"` and u/v supplied as `timedelta`s. I think passing u/v as timedeltas should _not_ be included with the (default) `angles="uv"`